src/Common/src/TypeSystem/Common/Utilities/GCPointerMap.Algorithm.cs

```diff
@@ -43,10 +43,9 @@ public static GCPointerMap FromStaticLayout(DefType type)
                 {
                     Debug.Assert(fieldType.IsValueType);
                     var fieldDefType = (DefType)fieldType;
-                    Debug.Assert(fieldDefType.ContainsGCPointers);
 
                     GCPointerMapBuilder innerBuilder =
-                        builder.GetInnerBuilder(field.Offset, fieldDefType.InstanceByteCount);
+                        builder.GetInnerBuilder(field.Offset, fieldDefType.InstanceFieldSize);
```

Comment:
I had unanswered questions about this change when it was in code review -

1. We have a high bar for type system related changes. Bug fixes need accompanying unit test (I can give you pointers if you haven't done it before)
2. There is very similar code for threadstatics. Why don't we need the change there?